src/ng/directive/select.js

```diff
@@ -80,6 +80,9 @@ var SelectController =
 
   // Tell the select control that an option, with the given value, has been added
   self.addOption = function(value, element) {
+    // Skip comment nodes, as they only pollute the `optionsMap`
+    if (element.prop('nodeType') === NODE_TYPE_COMMENT) return;
```

Comment:
That was my first "reaction", but then I thought: "What if `element[0]` is undefined ?"
So, instead of `element[0] && element[0].nodeType`, I went for `element.prop(...)` (I'm not even sure if the element passed to the linking function can be empty...)